infrastructure/src/main/java/org/corfudb/infrastructure/NettyServerRouter.java

```diff
@@ -143,7 +192,17 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) {
                         log.trace("Message routed to {}: {}", handler.getClass().getSimpleName(),
                                 msg);
                     }
-                    handlerWorkers.submit(() -> handler.handleMessage(m, ctx, this));
+                    handlerWorkers.submit(() -> {
+                        try {
+                            handler.handleMessage(m, ctx, this);
+                        } catch (Throwable t) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1181)